### DISCUSSION TITLE
Keycloak returning wrong Access Token after refresh.

### DISCUSSION THREAD

**Discussion by siangi** [keycloak/keycloak#28921] [Q&A]
Hello everyone

We ran into an issue with having multiple sessions on one client with the same realm/client. When a user1 logs in and doesn't terminate the session properly and then user2 logs in, both of them will have the same sid claim in the token. When user2 then uses the refresh token, an access token for user1 is returned.
This was tested with keycloak 23.0.4 and 24.0.2 with the standard browser authentication flow and the keycloak.js adapter.

Steps to reproduce:

1. Call keycloak.login(prompt: 'login)
2. User1 logs in and gets the proper access & refresh token (sid: 1234)
3. User1 doesn't log out.
4. Call keycloak.login(prompt: 'login')
5. Redirect to the password form
6. Restart authentication process (link next to username).
7. User2 logs in and gets the access & refresh token for user2 (also sid: 1234)
8. Call keycloak.updateToken() with the refresh token of user2
9. Keycloak returns an access token for user1

Shouldn't Keycloak create a new sid if the authentication process is restarted from scratch? Returning a token for the wrong user seems like a security issue.

**Comment by geckodesalpes** [comment]
We have noticed exactly the same issue, but we cannot figure out wether it is a Feature, a configuration issue or an actual bug.
We did not identified either any section in doc talking about it. This may be as well something issued form OIDC specifications, but this goes as well beyond my knowledges.

We have as well the feeling, that it may eventually be a security issue, from our current level of understanding.

Edit 1:
- Reverting KC team tag
- Add new findings

After a search in the CVE database, I found [the following one](https://www.cvedetails.com/cve/CVE-2023-6787/), meaning this is indeed a known security issue. The issue seems no more present on the main branch, and is claimed fixed apparently since version 22, which sounds odd as version 23 (and 24 but not tested personally) seems still affected by it.

**Comment by geckodesalpes** [comment]
The issue seems to be fixed in v25.0.2. I was at least not able to reproduce it as on previous versions.
